lib/write-type.js

```diff
@@ -52,29 +52,28 @@ function getWriteType(options) {
   }
 
   function number(encoder, value) {
-    var ivalue = value | 0;
     var type;
-    if (value !== ivalue) {
+    if (Math.floor(value) != value) {
       // float 64 -- 0xcb
       type = 0xcb;
       token[type](encoder, value);
       return;
-    } else if (-0x20 <= ivalue && ivalue <= 0x7F) {
+    } else if (-0x20 <= value && value <= 0x7F) {
       // positive fixint -- 0x00 - 0x7f
       // negative fixint -- 0xe0 - 0xff
-      type = ivalue & 0xFF;
-    } else if (0 <= ivalue) {
+      type = value & 0xFF;
+    } else if (0 <= value) {
       // uint 8 -- 0xcc
       // uint 16 -- 0xcd
       // uint 32 -- 0xce
-      type = (ivalue <= 0xFF) ? 0xcc : (ivalue <= 0xFFFF) ? 0xcd : 0xce;
+      type = (value <= 0xFF) ? 0xcc : (value <= 0xFFFF) ? 0xcd : (value <= 0xFFFFFFFF) ? 0xce : 0xcf;
```

Comment:
I think this, together with line 74 are the most important. All the others should be reversed.